### PR TITLE
135177 More lint fixes

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#13577

*Description of changes:*
- lint fixes
- app version = 4.2.2

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/interfaces/payment-interfaces/fees-interface.ts [hunk 1]
```diff
@@ -11,11 +17,6 @@ export interface FeesIF {
   total?: number
 }
 
-export interface TaxesIF {
```
Comment (severinbeauvais) [inline src/interfaces/payment-interfaces/fees-interface.ts:14]: 'TaxesIF' was used before it was defined

File: src/interfaces/payment-interfaces/fees-interface.ts [hunk 2]
```diff
@@ -1,4 +1,10 @@
 /** Interfaces for Fees from Payment API. */
+
+export interface TaxesIF {
+  pst: number
+  gst: number
+}
+
```
Comment (severinbeauvais) [inline src/interfaces/payment-interfaces/fees-interface.ts:7]: Fix for "'TaxesIF' was used before it was defined".

File: src/interfaces/stepper-interfaces/CreateShareStructure/create-share-structure-interface.ts [hunk 3]
```diff
@@ -17,6 +12,11 @@ export interface ShareClassIF {
   series?: ShareClassIF[]
 }
 
+export interface ShareStructureIF {
+  valid: boolean
+  shareClasses: ShareClassIF[]
+}
+
```
Comment (severinbeauvais) [inline src/interfaces/stepper-interfaces/CreateShareStructure/create-share-structure-interface.ts:19]: 'ShareClassIF' was used before it was defined

File: src/mixins/add-edit-org-person-mixin.ts [hunk 4]
```diff
@@ -502,13 +502,17 @@ export default class AddEditOrgPersonMixin extends Vue {
 
   // Event emitters
   @Emit('addEditPerson')
-  private emitPersonInfo (personInfo: OrgPersonIF): void {}
+  private emitPersonInfo (personInfo: OrgPersonIF): OrgPersonIF {
+    return personInfo
+  }
```
Comment (severinbeauvais) [inline src/mixins/add-edit-org-person-mixin.ts:507]: 'personInfo' is defined but never used

File: src/mixins/add-edit-org-person-mixin.ts [hunk 5]
```diff
@@ -502,13 +502,17 @@ export default class AddEditOrgPersonMixin extends Vue {
...
 
   @Emit('resetEvent')
   private emitResetEvent (): void {}
 
   @Emit('removePerson')
-  protected emitRemovePerson (activeIndex: number): void {}
+  protected emitRemovePerson (activeIndex: number): number {
+    return activeIndex
+  }
```
Comment (severinbeauvais) [inline src/mixins/add-edit-org-person-mixin.ts:515]: 'activeIndex' is defined but never used

File: src/services/legal-services.ts [hunk 6]
```diff
@@ -52,6 +52,7 @@ export default class LegalServices {
       .then(response => {
         // look for dissolution task
         const filing = response?.data?.tasks
+          // eslint-disable-next-line no-prototype-builtins
```
Comment (severinbeauvais) [inline src/services/legal-services.ts:55]: Do not access Object.prototype method 'hasOwnProperty' from target object

File: src/store/getters/state-getters.ts [hunk 7]
```diff
@@ -292,6 +292,7 @@ export const getBusinessContact = (state: StateIF): ContactPointIF => {
 }
 
 /** The Memorandum object. */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:295]: 'state' is defined but never used

File: tests/unit/Actions.spec.ts [hunk 8]
```diff
@@ -102,7 +102,6 @@ describe('Actions component', () => {
 describe('Emits error event if NR validation fails in file and pay', () => {
   let wrapper: any
   const { assign } = window.location
-  const effectiveDate = new Date(new Date().setDate(new Date().getDate() + 5))
```
Comment (severinbeauvais) [inline tests/unit/Actions.spec.ts:105]: 'effectiveDate' is assigned a value but never used

File: src/interfaces/payment-interfaces/fees-interface.ts (hunk 1 above)
Comment (severinbeauvais) [inline src/interfaces/payment-interfaces/fees-interface.ts:14]: 'TaxesIF' was used before it was defined

File: src/interfaces/stepper-interfaces/CreateShareStructure/create-share-structure-interface.ts [hunk 9]
```diff
@@ -1,8 +1,3 @@
-export interface ShareStructureIF {
```
Comment (severinbeauvais) [inline src/interfaces/stepper-interfaces/CreateShareStructure/create-share-structure-interface.ts:1]: 'ShareClassIF' was used before it was defined

File: src/mixins/add-edit-org-person-mixin.ts [hunk 10]
```diff
@@ -502,13 +502,17 @@ export default class AddEditOrgPersonMixin extends Vue {
 
   // Event emitters
   @Emit('addEditPerson')
-  private emitPersonInfo (personInfo: OrgPersonIF): void {}
```
Comment (severinbeauvais) [inline src/mixins/add-edit-org-person-mixin.ts:505]: 'personInfo' is defined but never used

File: src/mixins/add-edit-org-person-mixin.ts [hunk 11]
```diff
@@ -502,13 +502,17 @@ export default class AddEditOrgPersonMixin extends Vue {
...
+  private emitPersonInfo (personInfo: OrgPersonIF): OrgPersonIF {
+    return personInfo
+  }
 
   @Emit('resetEvent')
   private emitResetEvent (): void {}
 
   @Emit('removePerson')
-  protected emitRemovePerson (activeIndex: number): void {}
```
Comment (severinbeauvais) [inline src/mixins/add-edit-org-person-mixin.ts:511]: 'activeIndex' is defined but never used

File: src/router/router.ts [hunk 12]
```diff
@@ -14,6 +14,7 @@ export function getVueRouter () {
     // set base URL for Vue Router
     base: sessionStorage.getItem('VUE_ROUTER_BASE'),
     routes,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (severinbeauvais) [inline src/router/router.ts:17]: 'to' is defined but never used
'from' is defined but never used
'savedPosition' is defined but never used

File: src/services/legal-services.ts (hunk 6 above)
Comment (severinbeauvais) [inline src/services/legal-services.ts:55]: Do not access Object.prototype method 'hasOwnProperty' from target object

File: src/store/getters/state-getters.ts (hunk 7 above)
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:295]: 'state' is defined but never used

File: tests/unit/AddNameTranslation.spec.ts [hunk 13]
```diff
@@ -14,10 +14,6 @@ Vue.use(Vuelidate)
 const vuetify = new Vuetify({})
 const store = getVuexStore()
 
-function resetStore (): void {
```
Comment (severinbeauvais) [inline tests/unit/AddNameTranslation.spec.ts:17]: 'resetStore' is defined but never used

File: tests/unit/BusinessTypeConfirm.spec.ts [hunk 14]
```diff
@@ -1,5 +1,4 @@
 import Vue from 'vue'
-import flushPromises from 'flush-promises'
```
Comment (severinbeauvais) [inline tests/unit/BusinessTypeConfirm.spec.ts:2]: 'flushPromises' is defined but never used

File: tests/unit/CompleteResolution.spec.ts [hunk 15]
```diff
@@ -1,4 +1,3 @@
-import Vue from 'vue'
```
Comment (severinbeauvais) [inline tests/unit/CompleteResolution.spec.ts:1]: 'Vue' is defined but never used

File: tests/unit/DissolutionFirm.spec.ts [hunk 16]
```diff
@@ -1,6 +1,6 @@
 import Vuetify from 'vuetify'
 import { shallowWrapperFactory } from '../jest-wrapper-factory'
-import { mount, shallowMount } from '@vue/test-utils'
```
Comment (severinbeauvais) [inline tests/unit/DissolutionFirm.spec.ts:3]: 'shallowMount' is defined but never used

File: tests/unit/DissolutionReviewConfirm.spec.ts [hunk 17]
```diff
@@ -1,10 +1,7 @@
 import { shallowWrapperFactory, wrapperFactory } from '../jest-wrapper-factory'
-import { getVuexStore } from '@/store'
 import DissolutionReviewConfirm from '@/views/Dissolution/DissolutionReviewConfirm.vue'
 import { DissolutionResources } from '@/resources/'
 
-const store = getVuexStore()
```
Comment (severinbeauvais) [inline tests/unit/DissolutionReviewConfirm.spec.ts:6]: 'store' is defined but never used

File: tests/unit/IncorporationDateTime.spec.ts [hunk 18]
```diff
@@ -6,7 +6,7 @@ import { getLastEvent } from '../get-last-event'
 import { getVuexStore } from '@/store'
 
 // Utils
-import { createLocalVue, mount, Wrapper } from '@vue/test-utils'
```
Comment (severinbeauvais) [inline tests/unit/IncorporationDateTime.spec.ts:9]: 'Wrapper' is defined but never used

File: tests/unit/ListNameTranslations.spec.ts [hunk 19]
```diff
@@ -15,10 +15,6 @@ const vuetify = new Vuetify({})
 const store = getVuexStore()
 document.body.setAttribute('data-app', 'true')
 
-function resetStore (): void {
```
Comment (severinbeauvais) [inline tests/unit/ListNameTranslations.spec.ts:18]: 'resetStore' is defined but never used

File: tests/unit/RegAddEditOrgPerson.spec.ts [hunk 20]
```diff
@@ -29,7 +29,6 @@ const middleNameSelector = '.middle-name'
 const lastNameSelector = '.last-name'
 const confirmCheckboxSelector = '.confirm-checkbox'
 const orgNameSelector = '.org-name'
-const businessNumberSelector = '.business-number'
```
Comment (severinbeauvais) [inline tests/unit/RegAddEditOrgPerson.spec.ts:32]: 'businessNumberSelector' is defined but never used

File: tests/unit/RegistrationReviewConfirm.spec.ts [hunk 21]
```diff
@@ -1,10 +1,7 @@
-import { shallowWrapperFactory, wrapperFactory } from '../jest-wrapper-factory'
```
Comment (severinbeauvais) [inline tests/unit/RegistrationReviewConfirm.spec.ts:1]: 'wrapperFactory' is defined but never used

File: tests/unit/RegistrationReviewConfirm.spec.ts [hunk 22]
```diff
@@ -1,10 +1,7 @@
...
-import { getVuexStore } from '@/store'
+import { shallowWrapperFactory } from '../jest-wrapper-factory'
 import { RegistrationReviewConfirm } from '@/views'
 import { RegistrationResources } from '@/resources'
 
-const store = getVuexStore()
```
Comment (severinbeauvais) [inline tests/unit/RegistrationReviewConfirm.spec.ts:6]: 'store' is defined but never used

**Review by lambert-alex — APPROVED** [review]
LGTM

**Pull request merged**